airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/convert/SqlTypeToMssqlType.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.mssql.v2.convert
+
+import java.sql.Types
+
+enum class MssqlType(val sqlType: Int, val sqlStringOverride: String? = null) {
+    TEXT(Types.LONGVARCHAR),
+    BIT(Types.BOOLEAN),
+    DATE(Types.DATE),
+    BIGINT(Types.BIGINT),
+    DECIMAL(Types.DECIMAL, sqlStringOverride = "DECIMAL(18, 8)"),
+    VARCHAR(Types.VARCHAR, sqlStringOverride = "VARCHAR(MAX)"),
+    DATETIMEOFFSET(Types.TIMESTAMP_WITH_TIMEZONE),
+    TIME(Types.TIME),
+    DATETIME(Types.TIMESTAMP);
+
+    val sqlString: String = sqlStringOverride ?: name
+
+    companion object {
+        val fromSqlType: Map<Int, MssqlType> =
```

Comment:
man. I almost want to define a type just so we're not passing around raw Int values...
```kotlin
@JvmInline
value class JdbcType(val type Int)
```

Reply:
This is part of some aspects I was thinking of refactoring later on during the beta. The initial implementation started with a conversion to `SqlType` which we wired in, then I added a conversion to `MssqlType` which is the actual one we need. I want to collapse both layers to only deal with `MssqlType`.